contracts/oracles/accumulator/AccumulatorMultiOracle.sol

```diff
@@ -0,0 +1,128 @@
+
+    /**
+    @notice Set a source
+    @param baseId: base to set the source for
+    @param kindId: kind of oracle (example: chi/rate)
+    @param startRate: rate the oracle starts with
+    @param perSecondRate: secondly rate
+     */
+    function setSource(
+        bytes6 baseId,
+        bytes6 kindId,
+        uint256 startRate,
+        uint256 perSecondRate
+    ) external auth {
+        Accumulator memory source = sources[baseId][kindId];
+        require(source.accumulated == 0, "Source is already set");
+
+        sources[baseId][kindId] = Accumulator({
+            perSecondRate: perSecondRate,
+            accumulated: startRate,
+            lastUpdated: block.timestamp
+        });
+        emit SourceSet(baseId, kindId, startRate, perSecondRate);
+    }
+
+    /**
+    @notice Updates accumulation rate
+    
+    The accumulation rate can only be updated on an up-to-date oracle: get() was called in the
+    same block. See get() for more details
+     */
+    function updatePerSecondRate(
+        bytes6 baseId,
+        bytes6 kindId,
+        uint256 perSecondRate
+    ) external auth {
+        Accumulator memory source = sources[baseId][kindId];
+        require(source.accumulated != 0, "Source not found");
+
+        require(source.lastUpdated == block.timestamp, "stale accumulator");
+        sources[baseId][kindId].perSecondRate = perSecondRate;
+
+        emit PerSecondRateUpdated(baseId, kindId, perSecondRate);
+    }
+
+    /**
+     * @notice Retrieve the latest stored accumulated rate.
+     */
+    function peek(
+        bytes32 base,
+        bytes32 kind,
+        uint256
+    ) external view virtual override returns (uint256 accumulated, uint256 updateTime) {
+        Accumulator memory source = sources[base.b6()][kind.b6()];
+        require(source.accumulated != 0, "Source not found");
+
+        accumulated = source.accumulated;
+        require(accumulated > 0, "Accumulated rate is zero");
+
+        updateTime = block.timestamp;
+    }
+
+    /**
+    @notice Retrieve the latest accumulated rate from source, updating it if necessary.
+
+    Computes baseRate ^ (block.timestamp - creation timestamp)
+
+    pow() is not O(1), so the naive implementation will become slower as the time passes
+    To workaround that, each time get() is called, we:
+        1) compute the return value
+        2) store the return value in `accumulated` field, update lastUpdated timestamp
+
+    Becase we have `accumulated`, step 1 becomes `accumulated * baseRate ^ (block.timestamp - lastUpdated)
+     */
+    function get(
+        bytes32 base,
+        bytes32 kind,
+        uint256
+    ) external virtual override returns (uint256 accumulated, uint256 updateTime) {
+        Accumulator memory accumulator = sources[base.b6()][kind.b6()];
+        require(accumulator.accumulated != 0, "Source not found");
+
+        uint256 cycles = (block.timestamp - accumulator.lastUpdated);
```

Comment:
Take into account that there could be several requests in each block, and with the current implementation we would waste gas.

If `cycles == 0` you should skip the updating, and just return the stored value.

Also, `cycles` is a strange choice for the variable, `secondsAgo`, `secondsSinceUpdated`?